File: week1/윤여준/K번째수.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;

    for(auto v : commands) {
        int i = v[0], j = v[1], k = v[2];

        vector<int> temp;
        for(int l = i - 1; l < j; l++) temp.push_back(array[l]);

        sort(temp.begin(), temp.end());

        answer.push_back(temp[k-1]);
    }

    return answer;
}

File: week1/이다혜/K번째수.java
import java.util.Arrays;

public class 프로그래머스_K번째수 {
    public int[] solution(int[] array, int[][] commands) {
        int n = commands.length;
        int[] answer = new int[n];

        int idx = 0;
        for(int [] command : commands) {
            int s = command[0], e = command[1], k = command[2];

            int [] partialArr = new int[e-s+1];
            for(int i = s; i <= e; i++) {
                partialArr[i-s] = array[i-1];
            }

            Arrays.sort(partialArr);

            answer[idx++] = partialArr[k-1];
        }

        return answer;
    }
}
